core/opendaq/security/include/opendaq/authentication_provider_factory.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2022-2023 Blueberry d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+#include <coretypes/objectptr.h>
+#include <opendaq/authentication_provider_ptr.h>
+#include <opendaq/authentication_provider_impl.h>
+
+BEGIN_NAMESPACE_OPENDAQ
+
+/*!
+ * @ingroup opendaq_security
+ * @addtogroup opendaq_security_user Factories
+ * @{
+ */
+
+/*!
+ * @brief Creates an empty authentication provider without any user.
+ */
+inline AuthenticationProviderPtr AuthenticationProvider()
+{
+    AuthenticationProviderPtr obj(createWithImplementation<IAuthenticationProvider, StaticAuthenticationProviderImpl>(List<IUser>()));
+    return obj;
+}
+
+/*!
+ * @brief Creates an authentication provider out of static list of users.
+ * @param users List of user objects.
+ */
+inline AuthenticationProviderPtr StaticAuthenticationProvider(const ListPtr<IUser>& users)
```

Comment:
Any particular reason why we use inline factories? This means that implementation files (cpp or static libs) have to be used when creating providers, unless the object is header only, which is not.

Reply:
No particular reasons. I have moved the factories to cpp files.